receiver/m365receiver/factory.go

```diff
@@ -0,0 +1,97 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package m365receiver // import "github.com/observiq/observiq-otel-collector/receiver/m365receiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/observiq/observiq-otel-collector/receiver/m365receiver/internal/metadata"
+)
+
+const (
+	typeStr = "m365"
+)
+
+// NewFactory creates a factory for Microsoft Office 365 receiver.
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, metadata.Stability),
+		receiver.WithLogs(createLogsReceiver, metadata.Stability),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			CollectionInterval: 24 * time.Hour,
+		},
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			Timeout: 10 * time.Second,
+		},
+		MetricsBuilderConfig: metadata.DefaultMetricsBuilderConfig(),
+		Logs: &LogsConfig{
+			PollInterval:   5 * time.Minute,
```

Comment:
Is this a sensible default for this log receiver?

Reply:
Not too sure, Microsoft doesn't suggest a polling interval. That default was picked pretty arbitrarily. The most Microsoft says on the topic is in the screenshot below. In my limited testing, it seems like it can take up to an hour for logs to be reported, but I'm sure that can change as the load does. As for the default I'm open to a different one if you think it'd be better.
<img width="802" alt="Screenshot 2023-05-30 at 8 07 14 AM" src="https://github.com/observIQ/observiq-otel-collector/assets/122491662/84ae0f06-bb36-4c42-8c33-45f3805a9748">

Reply:
If it takes an hour for logs to reported, then a polling frequency of 5 minutes may be quite excessive. Perhaps a default of 15 minutes would be more appropriate. I can't really find any other information to go off of. What are your thoughts  @dehaansa ?

Reply:
While there may be an initial gap before logs are available, logs should then be somewhat consistently populating and available. 5 minutes seems like a reasonable number to me, to balance responsiveness & API usage.